backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java

```diff
@@ -110,20 +127,38 @@ public void getExperimentRunByName(
       GetExperimentRunByName request,
       StreamObserver<GetExperimentRunByName.Response> responseObserver) {
     try {
-      final var response =
-          futureExperimentRunDAO
-              .findExperimentRuns(
-                  FindExperimentRuns.newBuilder()
-                      .setExperimentId(request.getExperimentId())
-                      .addPredicates(
-                          KeyValueQuery.newBuilder()
-                              .setKey("name")
-                              .setValue(
-                                  Value.newBuilder().setStringValue(request.getName()).build())
-                              .setOperator(OperatorEnum.Operator.EQ)
-                              .setValueType(ValueTypeEnum.ValueType.STRING)
-                              .build())
-                      .build())
+      final var requestValidationFuture =
+          InternalFuture.runAsync(
+              () -> {
+                if (request.getName().isEmpty()) {
+                  String errorMessage =
+                      "ExperimentRun name not found in GetExperimentRunByName request";
+                  throw new InvalidArgumentException(errorMessage);
+                } else if (request.getExperimentId().isEmpty()) {
+                  String errorMessage = "Experiment ID not found in GetExperimentRunByName request";
+                  throw new InvalidArgumentException(errorMessage);
+                }
+              },
+              executor);
```

Comment:
As per the test cases we should have to validate request parameter first and here validation was missing so added it here